pinot-core/src/main/java/org/apache/pinot/core/query/pruner/ColumnValueSegmentPruner.java

```diff
@@ -238,6 +238,49 @@ private boolean pruneRangePredicate(IndexSegment segment, RangePredicate rangePr
     return false;
   }
 
+  /**
+   * For IN predicate, prune the segment based on:
+   * <ul>
+   *   <li>Column min/max value</li>
+   * </ul>
+   */
+  private boolean pruneInPredicate(IndexSegment segment, InPredicate inPredicate, Map<String, DataSource> dataSourceCache) {
```

Comment:
(nit) please consider adding more details to javadoc highlighting when the pruning won't happen if number of values is greater than threshold.

Reply:
done!